vm/starnix/starnix.go

```diff
@@ -231,14 +237,42 @@ func (inst *instance) connectToAdb(timeout time.Duration) error {
 			"connect",
 			fmt.Sprintf("127.0.0.1:%d", inst.port))
 		if err == nil && strings.HasPrefix(string(connectOutput), "connected to") {
+			if inst.debug {
+				log.Logf(0, "connected to adb server")
+			}
 			return nil
 		}
 		inst.runCommand("adb", "disconnect", fmt.Sprintf("127.0.0.1:%d", inst.port))
 		if inst.debug {
-			log.Logf(0, "adb connect failed")
+			log.Logf(1, "adb connect failed")
 		}
 		if time.Since(startTime) > timeout {
-			return fmt.Errorf("can't connect to ADB server")
+			return fmt.Errorf("can't connect to adb server")
+		}
+	}
+}
+
+func (inst *instance) restartAdbAsRoot() error {
+	startTime := time.Now()
+	for {
+		vmimpl.SleepInterruptible(3 * time.Second)
+		if inst.debug {
+			log.Logf(1, "attempting to restart adbd with root access")
+		}
+		err := inst.runCommand(
+			"adb",
+			"-s",
+			fmt.Sprintf("127.0.0.1:%d", inst.port),
+			"root",
+		)
+		if err == nil {
+			return nil
+		}
+		if inst.debug {
+			log.Logf(1, "adb root failed")
+		}
+		if time.Since(startTime) > (2 * time.Minute) {
```

Comment:
Nit: Maybe we should pass the timeout as an argument to the method, similar to `startAdbServerAndConnection` and `connectToAdb`?

Reply:
Sounds good -- I consolidated these into properties on the instance instead, since at least as a first pass I think it makes sense to have the same timeout and retry interval for all of the adb commands here, and one of them was within the `Copy` interface method whose signature was already defined. 

I set all the adb timeouts to 2m for now; it's generous, but booting the emulator takes long enough that I think it's reasonable to avoid restarting it whenever possible.